server/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java

```diff
@@ -153,9 +167,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(INGEST_TOOK, ingestTookInMillis);
         }
         builder.field(ERRORS, hasFailures());
+        boolean noItems = noItemsOnSuccess != null && noItemsOnSuccess;
+        boolean noFailuresAndNoItemsRequested = noItems && this.hasFailures() == false;
+        builder.field(ITEMS_OMITTED, noFailuresAndNoItemsRequested);
```

Comment:
maybe we can return `items` with `null` value instead of returning the `items_omitted` field?
It looks like we currently never return `null`, so we could use that.

Reply:
The third state for `items_omitted` adds more complexity to the users of the API IMO. Knowing the property is always there makes checking for it all the easier.